xbmc/cores/VideoPlayer/VideoPlayer.cpp

```diff
@@ -2439,6 +2439,18 @@ void CVideoPlayer::OnExit()
   if (!m_bAbortRequest)
     CLog::Log(LOGNOTICE, "VideoPlayer: eof, waiting for queues to empty");
 
+  bool updateStreamDetails = NeedUpdateStreamDetails(m_item.GetPath());
```

Comment:
instead of having a method "NeedUpdateStreamDetails", create one for this entire block that sets the details